src/frameworks/utils.ts

```diff
@@ -0,0 +1,23 @@
+import { readFile } from "fs/promises";
+import { join } from "path";
+
+/**
+ * Prints a warning if the build script in package.json
+ * contains anything other than allowedBuildScripts.
+ */
+export async function warnIfCustomBuildScript(
+  dir: string,
+  allowedBuildScripts: string[]
+): Promise<void> {
+  const packageJsonBuffer = await readFile(join(dir, "package.json"));
+  const packageJson = JSON.parse(packageJsonBuffer.toString());
+  const buildScript = packageJson.scripts?.build;
+
+  if (buildScript && !allowedBuildScripts.includes(buildScript)) {
+    console.warn(
```

Comment:
I'll see what folk think of this terminology

Reply:
I think this would be better wording

`WARNING: Your package.json contains a custom build that is being ignored. Only the ${framework} default build script (e.g, "${defaultBuildScripts[0]}") is respected. If you have a more advanced build process you should build a custom integration https://firebase.google.com/docs/hosting/express\n`